### DISCUSSION TITLE
Not updating state between two components.

### DISCUSSION THREAD

**Discussion by nhutruong-struct** [pmndrs/zustand#2091] [Q&A]
Hi community,
I've been using Zustand to update the state, but I'm not receiving state changes in another component. I've written a custom hook as follows:
![image](https://github.com/pmndrs/zustand/assets/101889353/d787b182-2e35-4c70-b016-19374a5ea118)
I want to pass the percentage of a loaded file from the UploadFileIcon component to the Processing component, but in the Processing component, I don't see any changes. I have a console.log in my useIfcModel hook, and it shows that the percentage value has changed. I'm not sure where I might be going wrong; I hope someone can help me.
![image](https://github.com/pmndrs/zustand/assets/101889353/a12f61a4-4a8c-44ad-a05e-95beced6157c)
![image](https://github.com/pmndrs/zustand/assets/101889353/ac5aa788-0e21-4df8-9b43-c8641fb1eadf)

**Comment by nhutruong-arent** [comment]
@dbritto-dev 
Thank you for your support. 
However, I've noticed in the Zustand documentation that it's possible to update state between components, and my context seems to be similar, but I'm still unable to update the state between components.
https://github.com/pmndrs/zustand#then-bind-your-components-and-thats-it

**Reply by dbritto-dev** [reply]
@nhutruong-arent my bad I was talking about something different in your case that should work I would love to help you but I need a codesandbox demo with your changes, just a basic example in order to replicate your issue
